### PR TITLE
Remove declarativeContent API usage to allow Firefox support

### PR DESCRIPTION
**Pull request by cameronkeif**
Refactored `background.js` for both extensions to remove the `declarativeContent` API usage (and accompanying permission), which allows the extensions to be ported to Firefox.

The result of testing the packed extension via [https://www.extensiontest.com](https://www.extensiontest.com/test/236f76a0-9d3f-11eb-b723-bb1813f63e77) (attached is the NBA result, but NHL is identical).

firefox-compatibility

Firefox demo

firefox

Chrome regression test

chrome

### CODE REVIEW

**Comment by magnusbakken** [comment]
Dude, this is awesome! I've been meaning to do this myself as I mentioned in #8. Looks like the changes aren't too big. I'll do some testing myself later today or tomorrow.

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "name": "espn-fantasy-autopick",
-    "version": "3.0.3",
-    "description": "A Chrome extension that lets you automatically add active players to the current roster in an ESPN NBA/NHL fantasy league.",
+    "version": "3.0.4",
```
Comment (magnusbakken) [inline package.json]: You can leave the version numbers as they are. I'll increment them in a separate commit for git tagging purposes.

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
+    "description": "An extension that lets you automatically add active players to the current roster in an ESPN NBA/NHL fantasy league.",
```
Comment (magnusbakken) [inline package.json]: Maybe "A browser extension that lets you..."?

File: src/nba/manifest.json [hunk 3]
```diff
@@ -1,10 +1,9 @@
 {
     "name": "ESPN NBA Fantasy Team Auto Setup",
     "description": "Automatically add active players to the current roster in an ESPN NBA fantasy league.",
-    "version": "3.0.3",
+    "version": "3.0.4",
```
Comment (magnusbakken) [inline src/nba/manifest.json]: Same note about the version number.

File: src/nhl/manifest.json [hunk 4]
```diff
@@ -1,10 +1,9 @@
 {
     "name": "ESPN NHL Fantasy Team Auto Setup",
     "description": "Automatically add active players to the current roster in an ESPN NHL fantasy league.",
-    "version": "3.0.3",
+    "version": "3.0.4",
```
Comment (magnusbakken) [inline src/nhl/manifest.json]: Same note about the version number.

**Review by magnusbakken — COMMENTED** [review]
Looks good! I'm planning to do something smarter to reduce the duplication between the different sports, so I think ultimately there will be a single background.js that takes the URL from some variable, but this is fine for now.

File: package.json (hunk 1 above)
Comment (cameronkeif) [reply package.json]: Ah cool, I will keep that in mind. Reverted back to `3.0.3` 0d5ae90fa17bc30f95756a3224f4cdb75d8c48d3

File: package.json (hunk 2 above)
Comment (cameronkeif) [reply package.json]: Done in 0d5ae90fa17bc30f95756a3224f4cdb75d8c48d3

File: src/nba/background.js [hunk 5]
```diff
@@ -1,16 +1,9 @@
...
-                conditions: [
-                    new chrome.declarativeContent.PageStateMatcher({
-                        pageUrl: { hostEquals: 'fantasy.espn.com', pathPrefix: '/basketball/team' },
-                    })
-                ],
-                actions: [ new chrome.declarativeContent.ShowPageAction() ]
-            }
-        ]);
-    });
+chrome.tabs.onUpdated.addListener(function(tabId, changeInfo, tab) {
```
Comment (magnusbakken) [inline src/nba/background.js:1]: I think this part should be changed as follows:

    chrome.tabs.onUpdated.addListener(function (tabId, changeInfo, tab) {
        if (changeInfo.status === 'complete') {
            if (tab.url && tab.url.match(/fantasy.espn.com\/basketball\/team/)) {
                chrome.pageAction.show(tabId);
            } else {
                chrome.pageAction.hide(tabId);
            }
        }
    });

And of course similarly in the NHL code.

With the current code, the extension popup normally doesn't show up for me. If I understand the logic of this correctly, we need to show the page action when the URL matches and hide it when it doesn't, but only when the status is "complete". Sometimes you get other messages where the status is "progress" or simply undefined.

**Review by magnusbakken — CHANGES_REQUESTED** [review]
After testing I've found that the auto-setup still works, but at least for me, the extension popup doesn't work anymore. I think I have the solution, but you'll have to verify that it works on your end as well.

File: src/nba/background.js (hunk 5 above)
Comment (cameronkeif) [reply src/nba/background.js:1]: What is the "extension popup"? I've only seen the buttons when I have used the extension so I might have missed that part -- what should I see, and how is it triggered?

Comment (magnusbakken) [reply src/nba/background.js:1]: The popup you get when you click the extension icon while on the clubhouse page:
image

With the current code it looks like this:
image

Comment (magnusbakken) [reply src/nba/background.js:1]: Hey @cameronkeif. Is my explanation clear? I expect that most people never use the popup page, but it does need to be there, at least for the "Set up for N days" feature.

Comment (cameronkeif) [reply src/nba/background.js:1]: Yes it is clear - sorry, I just haven't been able to work on this lately.

Comment (magnusbakken) [reply src/nba/background.js:1]: Ah, no worries. There's obviously no rush here. The NBA and NHL fantasy seasons are drawing to a close at this point anyway, so a possible Firefox version of the extension likely won't see any installs until the start of the next season.

**Review by magnusbakken — APPROVED** [review]

**Pull request merged**